api/app_analytics/views.py

```diff
@@ -88,7 +88,9 @@ def get_fields(self):  # type: ignore[no-untyped-def]
 
             def save(self, **kwargs: typing.Any) -> None:
                 request = self.context["request"]
-                for feature_name, evaluation_count in self.validated_data.items():
+                # validated_data splits out request body with '.' in feature name (e.g a.b.c).
+                # Instead, it's safe to use self.data as keys are not altered.
```

Comment:
This comment doesn't make sense to me. Surely we want the "." in the feature name. Can we be more explicit here about what the difference is between `validated_data` and `data` in this scenario?

Also, is there a neater way to solve this in the serializer rather than the view? I wonder if skipping validated_data here might lead to other problems down the line.

Reply:
Sorry for the late response.
Sure. I can update the comment stating the diff bw data and validated_data. 

In this particular scenario, the validated_data do not hold anything additional and it is safe to use the plain `data` since validation is successful at this point

Another solution is to leverage [SkipField Exception](https://github.com/encode/django-rest-framework/blob/e045dc465270c18689dba4a970378cd9744e57b6/rest_framework/serializers.py#L508). I am actually open to suggestions here as to what can be done.